### ISSUE TITLE
Improper dependency error handling causes hang

### ISSUE THREAD

**Issue by hategan** [Parsl/parsl#1285]
The following code completes with an exception:
```python
@python_app
def f(x):
    raise Exception('Ooops2')
    return 1

@python_app
def g():
    # raise Exception('Ooops1')
    return 1

result = f(g()).result()
```

That is expected. The following code hangs:
```python
@python_app
def f(x):
    raise Exception('Ooops2')
    return 1

@python_app
def g():
    raise Exception('Ooops1')
    return 1

result = f(g()).result()
```

This is because in the `else` clause of _launch_if_ready, the relevant callbacks are not invoked. If the following line is added at the end of said `else` clause,

```python
self.tasks[task_id]['app_fu'].parent_callback(fu)
```

then the second snippet correctly completes with a dependency error.

**Comment by benclifford** [comment]
I think this is #1257 #1259 -  #1259 was only merged a few minutes ago. I think the tests added in 1259 should correspond to your test case above. I'm not at a terminal to verify though.

**Comment by benclifford** [comment]
fixed by #1259
